pkg/middleware/middleware_test.go

```diff
@@ -0,0 +1,122 @@
+package middleware_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/weaveworks/weave-gitops/pkg/middleware"
+	"github.com/weaveworks/weave-gitops/pkg/services/auth"
+	"github.com/weaveworks/weave-gitops/pkg/services/auth/authfakes"
+)
+
+var (
+	jwtClient      *authfakes.FakeJWTClient
+	defaultHandler http.HandlerFunc
+)
+
+var _ = Describe("WithProviderToken", func() {
+	_ = BeforeEach(func() {
+		jwtClient = &authfakes.FakeJWTClient{
+			VerifyJWTStub: func(s string) (*auth.Claims, error) {
+				return &auth.Claims{
+					ProviderToken: "provider-token",
+				}, nil
+			},
+		}
+
+		defaultHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+	})
+
+	It("does nothing when no token is passed", func() {
+		midware := middleware.WithProviderToken(jwtClient, defaultHandler)
+
+		req := httptest.NewRequest(http.MethodGet, "http://www.foo.com/", nil)
+		res := httptest.NewRecorder()
+		midware.ServeHTTP(res, req)
+
+		Expect(res.Result().StatusCode).To(Equal(http.StatusOK))
+		Expect(jwtClient.VerifyJWTCallCount()).To(Equal(0))
+	})
+
+	It("extracts JWT token from the header", func() {
+		midware := middleware.WithProviderToken(jwtClient, defaultHandler)
+
+		req := httptest.NewRequest(http.MethodGet, "http://www.foo.com", nil)
+		req.Header.Add("Authorization", "token my-jwt-token")
+
+		res := httptest.NewRecorder()
+
+		midware.ServeHTTP(res, req)
+
+		Expect(jwtClient.VerifyJWTArgsForCall(0)).To(Equal("my-jwt-token"))
+	})
+
+	It("returns forbidden when token is no valid", func() {
+		jwtClient.VerifyJWTStub = func(s string) (*auth.Claims, error) {
+			return nil, auth.ErrUnauthorizedToken
+		}
+
+		midware := middleware.WithProviderToken(jwtClient, defaultHandler)
+		req := httptest.NewRequest(http.MethodGet, "http://www.foo.com", nil)
+		req.Header.Add("Authorization", "token my-jwt-token")
+
+		res := httptest.NewRecorder()
+
+		midware.ServeHTTP(res, req)
+
+		Expect(res.Result().StatusCode).To(Equal(http.StatusForbidden))
```

Comment:
Nit: Would `http.StatusUnauthorized` be more appropriate?

Reply:
yes, I'll change it